core/src/main/scala/ai/h2o/sparkling/backend/api/rdds/RDDsServlet.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.backend.api.rdds
+
+import ai.h2o.sparkling.H2OFrame
+import ai.h2o.sparkling.backend.api.{GETRequestBase, POSTRequestBase, ServletRegister}
+import ai.h2o.sparkling.utils.SparkSessionUtils
+import javax.servlet.Servlet
+import javax.servlet.http.HttpServletRequest
+import org.apache.spark.h2o.{H2OConf, H2OContext}
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.ScalaReflection
+import org.apache.spark.sql.types.{StructField, StructType}
+import water.exceptions.H2ONotFoundArgumentException
+
+/**
+  * This servlet class handles requests for /3/RDDs endpoint
+  */
+private[api] class RDDsServlet extends GETRequestBase with POSTRequestBase {
+  private lazy val hc = H2OContext.ensure()
+  private lazy val sc = hc.sparkContext
+
+  override def handlePostRequest(request: HttpServletRequest): Any = {
+    request.getRequestURI match {
+      case s if s.matches(toScalaRegex("/3/RDDs/*/h2oframe")) =>
+        val parameters = RDDToH2OFrame.RDDToH2OFrameParameters.parse(request)
+        parameters.validate()
+        toH2OFrame(parameters.rddId, parameters.h2oFrameId)
+      case s if s.matches(toScalaRegex("/3/RDDs/*")) =>
+        val parameters = RDDInfo.RDDInfoParameters.parse(request)
+        parameters.validate()
+        getRDD(parameters.rddId)
+      case invalid => throw new H2ONotFoundArgumentException(s"Invalid endpoint $invalid")
+    }
+  }
+
+  override def handleGetRequest(request: HttpServletRequest): Any = {
+    request.getRequestURI match {
+      case "/3/RDDs" =>
```

Comment:
Will this work with a context path?

Reply:
Good point, will add test as well

Reply:
Good catch, did not work, but should be fine now + tests for changed endpoints now run with context path